addons/mail_connector/models/mail_connector.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class MailConnector(models.Model):
+    _name = 'mail.connector'
+
+    @api.model
+    def _default_conntectors_type(self):
+        return [('none', 'None')]
+
+    name = fields.Char(string="Name", required=True)
+    channel_id = fields.Many2one('mail.channel', string="Channel", required=True)
+    connect_type = fields.Selection('_default_conntectors_type', string="Connector Type")
```

Comment:
no need to define function.. you can use selection_add attribute to overrride only selection attribute..